device/alpaka/src/clusterization/clusterization_algorithm.cpp

```diff
@@ -0,0 +1,136 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "traccc/alpaka/clusterization/clusterization_algorithm.hpp"
+
+#include "../utils/barrier.hpp"
+#include "../utils/utils.hpp"
+
+// Project include(s)
+#include "traccc/clusterization/device/ccl_kernel.hpp"
+
+// System include(s).
+#include <algorithm>
+
+namespace traccc::alpaka {
+
+struct CCLKernel {
+    template <typename TAcc>
+    ALPAKA_FN_ACC void operator()(
+        TAcc const& acc, const cell_collection_types::const_view cells_view,
+        const cell_module_collection_types::const_view modules_view,
+        const device::details::index_t max_cells_per_partition,
+        const device::details::index_t target_cells_per_partition,
+        measurement_collection_types::view measurements_view,
+        vecmem::data::vector_view<unsigned int> cell_links) const {
```

Comment:
Generally, this setup is fine with me. I'm just thinking out loud here.
  - Would it not be nicer / more symmetric to call this something like `traccc::alpaka::kernels::ccl_kernel`? That's the sort of naming that we went with in `traccc::cuda`.
  - I think Alpaka allows such structs to carry member variables as well. Or is it not flexible enough for that? :thinking: I was just wondering, since you pass an actual object instance to `alpaka::exec` and not just the type of the struct, could some of these (many) arguments be delegated to member variables? Technically, the "configuration variables" would seem like appropriate members, while the data views indeed look more understandable as operator arguments. :thinking:

Reply:
> Would it not be nicer / more symmetric to call this something like traccc::alpaka::kernels::ccl_kernel? That's the sort of naming that we went with in traccc::cuda.

I'd originally tried to split them up, so they were obviously distinct, but having them mirror the CUDA / actual kernel names could be nicer too. Regardless, I'll look at fixing that up in the follow up finding / fitting PR, just so I can change them all in one go.

> I think Alpaka allows such structs to carry member variables as well. Or is it not flexible enough for that? 🤔 I was just wondering, since you pass an actual object instance to alpaka::exec and not just the type of the struct, could some of these (many) arguments be delegated to member variables? Technically, the "configuration variables" would seem like appropriate members, while the data views indeed look more understandable as operator arguments. 🤔

Possibly! I think the style shown here matches the Alpaka-recommended way at least, but it may be better for our needs to swap....I can do some testing soon and see, and when I look at renaming the kernels, I can see if swapping lots of args to member variables makes sense.